Get ROCm building again on master

Jorghi12 opened the pull request:
Big pile of fixes for ROCm build on master!

Jorghi12 committed:
Update ROCm Docker to install hcrng & hcsparse

These two packages are necessary for building PyTorch AMD.

Jorghi12 committed:
Create all folders.

ezyang committed:
ROCM nightly build

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Fix mkdir

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

bddppq commented on docker/caffe2/jenkins/common/install_rocm.sh:
```diff
@@ -2,8 +2,18 @@
+# it ignores all values of ROCM_VERSION which are not nightly.  Ugh!
 [ -n "$ROCM_VERSION" ]
 
+install_hip_nightly() {
```
what does this install? will it override stuffs installed in install_ubuntu?

ezyang replied:
Yes, it builds and then installs some new deb files.

ezyang committed:
nightly hipBLAS

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
More periods.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Say yes to all prompts

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Jorghi12 committed:
Build PyTorch AMD

ezyang committed:
Actually install the nightlies

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Enable rocm build

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Patch Thrust

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
switch to nightly

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Don't install mkl/mkl-devel

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Hand apply THCTensorMathReduce patch

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Clear pytorch_amd folder before build

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Use correct version of Python

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
More crimes

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
sudo says

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
chmod it

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
preserve exe

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
More shenanigans

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
3.6

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
stuff

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
more shenanigans

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Jorghi12 committed:
Updates to PyTorch AMD

Jorghi12 committed:
Fix redefinition

Jorghi12 committed:
Renamed from hiprng to hcrng since the soname is different now.

Jorghi12 committed:
Updated script to handle different versions of PyThon.

Jorghi12 committed:
Removed unnecessary encoding param

Jorghi12 committed:
Using the updated patch file from sending_pr

ezyang committed:
Stop updating alternatives.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
git am doesn't work

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Jorghi12 committed:
Now using exec to keep the current process alive

Jorghi12 committed:
Merge branch 'Jorghi12-patch-1' of github.com:pytorch/pytorch into Jorghi12-patch-1

Jorghi12 committed:
Now doing the pytorch build in place.

Jorghi12 committed:
Doing transpilation in place. Not using caffe2_gpu at the moment.

Commit:
Merge branch 'master' into Jorghi12-patch-1

ezyang commented:
@pytorchbot retest this please

Jorghi12 committed:
Adding indentation to file and specializing to py2

Jorghi12 committed:
Merge branch 'Jorghi12-patch-1' of github.com:pytorch/pytorch into Jorghi12-patch-1

ezyang committed:
change the config again

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Try to unwedge the installer first

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Commit:
Update install_rocm.sh

Very hacky fix.

ezyang committed:
Set HIP_PLATFORM in script too

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Can't WERROR this dunananana

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Commit:
Set AMDGPU Target

When building on the ROCm stack for our docker images, compile to use gfx900, which specifies the architecture version.

ezyang committed:
track master change

ezyang commented on cmake/Dependencies.cmake:
```diff
@@ -475,7 +475,7 @@ if(USE_ROCM AND NOT BUILD_CAFFE2)
 
  # Link with HIPCC https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/hip_porting_guide.md#linking-with-hipcc
- SET(CMAKE_CXX_LINK_EXECUTABLE ${HIP_HIPCC_EXECUTABLE})
+ # SET(CMAKE_CXX_LINK_EXECUTABLE ${HIP_HIPCC_EXECUTABLE})
```
Next time, leave a comment saying what's going on with the commented out line

ezyang commented on tools/amd_build/patches/a_aten_src_THC_generic_THCTensorRandom.cu.patch:
```diff
@@ -1,13 +1,13 @@
      generate_random_64<<<NUM_BLOCKS, BLOCK_SIZE, 0, THCState_getCurrentStream(state)>>>(
 -        gen->state.gen_states, size, data, min_val, range);
-+        gen->state.gen_states, static_cast<int>(size), data, static_cast<std::conditional<std::is_same<real,float>::value, float, double>::type>(min_val), static_cast<std::conditional<std::is_same<real,float>::value, float, double>::type>(range));
++        gen->state.gen_states, static_cast<int>(size), data, min_val, range);
```
Let's try to get rid of this patch in the near future

Jorghi12 committed:
Update Dependencies.cmake

I made sure not to require hipsparse & hiprng when building Caffe2 since their docker image doesn't have it (and they don't require them at the moment).
